src/reducers/versions.spec.tsx

```diff
@@ -616,4 +619,237 @@ describe(__filename, () => {
+        expect(hunk).toHaveProperty('changes');
+      });
+    });
+
+    it('creates changes from external diff changes', () => {
+      const version = {
+        ...fakeVersion,
+        file: {
+          ...fakeVersion.file,
+          diff: [fakeExternalDiff],
+        },
+      };
+
+      const diff = _createInternalDiffs({ version })[0];
+
+      const firstHunk = diff.hunks[0];
+      const firstExternalHunk = fakeExternalDiff.hunks[0];
+
+      expect(firstHunk.changes).toHaveLength(firstExternalHunk.changes.length);
+      firstHunk.changes.forEach((change, index) => {
+        const externalChange = firstExternalHunk.changes[index];
+
+        expect(change).toHaveProperty('content', externalChange.content);
+        expect(change).toHaveProperty('type', externalChange.type);
+        expect(change).toHaveProperty(
+          'oldLineNumber',
+          externalChange.old_line_number,
+        );
+        expect(change).toHaveProperty(
+          'newLineNumber',
+          externalChange.new_line_number,
+        );
+
+        // These props will be tested later.
+        expect(change).toHaveProperty('lineNumber');
+        expect(change).toHaveProperty('isDelete');
+        expect(change).toHaveProperty('isInsert');
+        expect(change).toHaveProperty('isNormal');
+      });
+    });
+
+    it('creates "delete" changes', () => {
+      const type = 'delete';
+      const oldLineNumber = 123;
+      const version = createVersionWithChange({
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        old_line_number: oldLineNumber,
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        new_line_number: oldLineNumber + 1,
+        type,
+      });
+
+      const diff = _createInternalDiffs({ version })[0];
+      const change = diff.hunks[0].changes[0];
+
+      expect(change).toHaveProperty('type', type);
+      expect(change).toHaveProperty('isDelete', true);
+      expect(change).toHaveProperty('isInsert', false);
+      expect(change).toHaveProperty('isNormal', false);
+      expect(change).toHaveProperty('lineNumber', oldLineNumber);
+    });
+
+    it('creates "insert" changes', () => {
+      const type = 'insert';
+      const newLineNumber = 123;
+      const version = createVersionWithChange({
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        old_line_number: newLineNumber - 1,
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        new_line_number: newLineNumber,
+        type,
+      });
+
+      const diff = _createInternalDiffs({ version })[0];
+      const change = diff.hunks[0].changes[0];
+
+      expect(change).toHaveProperty('type', type);
+      expect(change).toHaveProperty('isDelete', false);
+      expect(change).toHaveProperty('isInsert', true);
+      expect(change).toHaveProperty('isNormal', false);
+      expect(change).toHaveProperty('lineNumber', newLineNumber);
+    });
+
+    it('creates "normal" changes', () => {
```

Comment:
Nit: You have individual tests for each of the change types, but there is no test that deals with a diff with multiple changes, and verifies that each of them are properly translated and the resulting InternalDiff is accurate. Maybe we don't need that, but it feels to me like something is missing.

Reply:
In `creates changes from external diff changes`, we iterate over a set of `changes` so we have that. Given that the test case highlighted above (and the similar ones) always assert the first `change`, we can conclude that if it works for the first one and we have multiple changes, the same logic will be applied to all changes.